app/concerns/assignable_types.rb

```diff
@@ -11,37 +11,21 @@ def short_type
 
   def validate_type
     errors.add(:type, "cannot be changed once an instance has been created") if type_changed? && !new_record?
-    errors.add(:type, "is not a valid type") unless self.class.valid_type?(type)
+    errors.add(:type, "is not a valid type") unless valid_type?
   end
 
   module ClassMethods
-    def load_types_in(module_name, my_name = module_name.singularize)
-      const_set(:MODULE_NAME, module_name)
-      const_set(:BASE_CLASS_NAME, my_name)
-      const_set(:TYPES, Dir[Rails.root.join("app", "models", module_name.underscore, "*.rb")].map { |path| module_name + "::" + File.basename(path, ".rb").camelize })
-    end
-
     def types
-      const_get(:TYPES).map(&:constantize)
-    end
-
-    def valid_type?(type)
-      const_get(:TYPES).include?(type)
+      Agent.descendants
```

Comment:
I don't think this will be compatible with #1366 and https://github.com/cantino/huginn_agent/pull/2, it would require to completely rethink and rewrite the Agent registration. (https://github.com/cantino/huginn_agent/pull/2/files/f22b823ef0b34c448accb01137ae917c28551746#r65042818)